Makefile-Tasking-Tricore/custom_compiler_yaml_tasking.py

```diff
@@ -0,0 +1,70 @@
+output_filename = "custom-compiler-tasking-generated.yaml"
+tasking_install_folder = "C:\\Program Files\\TASKING\\TriCore v6.3r1"
+tasking_platform_suffix = ["51", "arm", "mcs", "pcp", "tc"]  # TriCore, ARM
+
+c_additional_compile_switches = "--iso=99"  # example, may be empty
+
+cpp_additional_compile_switches = "--c++=11"  # example, may be empty
+
+import os, subprocess, tempfile, re
+
+
+def extract_defines(compiler_bin, src_name):
+    result = subprocess.run([compiler_bin, "-El", src_name], stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
+    if result.returncode != 0:
+        return None
+    all_defines = result.stdout.splitlines()
+    filtered_defines = filter(lambda s: s.split()[1] not in [b'__FILE__', b'__LINE__', b'__DATE__', b'__TIME__', ],
+                              all_defines)
+    str_defines = b'\n'.join(filtered_defines).decode("ascii")
+    return str_defines
+
+
+with open(output_filename, "wt") as yaml_file:
+    open("__empty.c", "wt")
```

Comment:
Something like this is probably more convenient (don't need the "finally" block), and more robust:

```
with tempfile.NamedTemporaryFile(suffix=".c") as fc, tempfile.NamedTemporaryFile(suffix=".cpp") as fcpp:
   .. extrat_defines(... fc.name ...)
```

Reply:
I tried that. Tasking compiler fails to open those temp files.